services/systemd/espeakup.service.in

```diff
@@ -6,10 +6,9 @@ After=systemd-udev-settle.service sound.target
 
 [Service]
 Type=forking
-EnvironmentFile=@sysconfdir@/conf.d/espeakup.conf
```

Comment:
Yes, drop-in files are better, but that's exactly what you can get with using `EnvironmentFile-=@sysconfdir@/conf.d/espeakup.conf` (`-=` instead of just `=`). Surely we could implement support for a configuration file in espeakup, but should we really do this in addition to argument parsing while we can easily achieve the same with `EnvironmentFile`?

Reply:
well, yesterday I and @williamh discussed just that.
we have come to the conclusion that it is better to follow the [recommendations of systemd](https://unix.stackexchange.com/questions/418851/location-to-place-systemd-service-environmentfile-when-creating-debian-package) itself in this regard. Besides, it seems to me that the mechanism of drop-in files really looks better.

> we could implement support for a configuration file in espeakup.

I also think about this option. if we will introduce more configuration
we should do that.

Reply:
Mmm, that discussion is about avoiding the use of `EnvironmentFile`, not avoiding the use of `Environment` itself, which can be configured with drop-in files as the systemd purists seem to like.

Also, their calling `/etc/default` or `/etc/sysconfig` "madness" looks strange when they are actually doing the same with drop-in files..

Reply:
oh now i understand. you want to use systemd environment directive.

but as far as I know espeakup itself does not read any environment variables.
so you still need to modify exec so isn't it better to change exec = with drop-in files?

Reply:
I don't "want" anything particular technically, I want something that is easy for users to configure. An existing file to modify and be done is easy. A file you have to create in a specific place is not really.

> as far as I know espeakup itself does not read any environment variables so you still need to modify exec

Yes, but that defines a variable that can be configured

> isn't it better to change exec = with drop-in files?

The content to be put would be very technical with that approach.

Reply:
ok... we can do something like this

```
Environment=default_voice=
ExecStart=@bindir@/espeakup --default-voice=${default_voice}
```

then users can create drop-in file with `systemctl edit`.

I really don't mind returning the config file at all. Until yesterday's comment by @williamh , I didn't think they were bad at all.

Reply:
```environment=``` is probably the way to go. ```@sysconfdir@/conf.d``` is where openrc configuration files go, so it is not a good idea to put systemd configuration files there.

Reply:
@sthibaul what do you think now?